### PR TITLE
Fix for  STS-4067

### PR DESCRIPTION
**Pull request by kdvolder**
getting started guide has wrong JDK set

See https://issuetracker.springsource.com/browse/STS-4067

### CODE REVIEW

File: complete/pom.xml [hunk 1]
```diff
@@ -27,6 +27,14 @@
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
             </plugin>
+            <plugin>
```
Comment (dsyer) [inline complete/pom.xml]: Because of the parent you don't need all this. You just need a property java.version=1.7.

Comment (dsyer) [inline complete/pom.xml]: Because of the parent you don't need all this: just a property java.version=1.7.

**Comment by gregturn** [comment]
Should we examine the template project found at https://github.com/spring-guides/getting-started-guides as well?

**Comment by dsyer** [comment]
@gregturn yes, probably. I don't know if something changed in eclipse, but I have issues now as well that I don't remember from a year ago (the compiler defaults to 1.5 and then tge project won't compile). Fixing the java.version fixes it for me.

**Comment by kdvolder** [comment]
Made the change to my PR as suggested by Dave (i.e. use 'java.version' property instead of configuring maven-compiler plugin).

**Comment by kdvolder** [comment]
> (the compiler defaults to 1.5 and then tge project won't compile). 

I think the compiler and JRE container will default to what is defined as the global default in one's workspace. (Because it isn't specified by the project itself). So possibly the difference between what you see now and a year ago is something to do with your workspace defaults.

It could be that I'm wrong about this and if that is the case maybe it warrants a closer look at why you are unexpectedly getting 1.5 as a default.

**Comment by gregturn** [comment]
I chatted with @martinlippert yesterday and updated all guides using a newer JDK with the proper maven/gradle settings. He said this would help STS.

**Comment by kdvolder** [comment]
Greg... not sure where or what you fixed. But I just tested this guide in STS and it still has the same problem. I do not see this:

	<properties>
	  <java.version>1.7</java.version>
	</properties>

in the pom.

Same with gradle import as well.

**Comment by gregturn** [comment]
Martin alerted me that tomcat 8 requires JDK 7 so I didn't update the right ones. When I frt back from PTO I'll craft a more comprehensive solution.

**Comment by gregturn** [comment]
Resolved via:

*  https://github.com/spring-guides/gs-rest-service/commit/f9196f354a3684eca81f8304a2080ab0cadd9a62 - upgrade maven builds to Java 7
* https://github.com/spring-guides/gs-rest-service/commit/4feb3e84550103cdeee5d7ae0b6e9a99e7ac0420 - upgrade gradle builds to Java 7
* https://github.com/spring-guides/gs-rest-service/commit/143e1b4c0f3636211473d3bb7912fb5979554ac4 - upgrade guide itself to indicate minimum Java 7

**Comment by kdvolder** [comment]
I still don't see this in the gradle build:

```
eclipse {
    classpath {
         containers.remove('org.eclipse.jdt.launching.JRE_CONTAINER')
         containers
'org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7'
    }
}
```

Which means, I think, you will still have the wrong JRE container when
imported to Eclipse.

**Comment by kdvolder** [comment]
PS: why you need that ugly piece of code in the gradle build is explained
here: https://issues.gradle.org/browse/GRADLE-3100

**Comment by gregturn** [comment]
That's atrocious. Why can't we submit a patch to Gradle? Maven doesn't do this. I don't want to maintain Java 1.7 in two places in a build file.

**Comment by kdvolder** [comment]
> Why can't we submit a patch to Gradle?

Maybe we can, if someone feels up to the task of fixing Gradle, even if we
do, won't fix the current problem with the guide until the patch is
accepted and released in a new version of Gradle.

Adding this 'cruft' to the build script isn't nice, but at least people
importing our guides don't get into "wrong JRE trouble" importing our
guides with Gradle.

It is a bit disappointing that Gradle folks don't respond to bugs like
this, and contrary to what they say publically (that they care about Gradle
+ Eclipse integration, and things will get better), I generally still don't
get a sense they care much at all.

Is it really up to us to fix bugs these kinds of bugs & problems in their
code?

**Comment by gregturn** [comment]
I don't like covering for Gradle. It's more maintenance costs for us.

If other projects (Spring Framework as one of many) don't have this workaround, why is up to us to put it in? Instead, people can flip the JDK setting inside Eclipse and then complain that Gradle doesn't integrate nicely with Gradle. Let the community see the issue with Gradle instead of sticking us with the covering for them.

**Pull request closed without merging**